components/connectkit/modals/connect-modal/scenes/input-email-to-register-1.tsx

```diff
@@ -0,0 +1,147 @@
+import { TextInput } from "../components/text-input";
+import { Field } from "../components/field";
+import { EmailIcon, PasswordIcon, RegisterIcon } from "../components/icons";
+import { NextStepButton } from "../components/next-step-button";
+import { CodeInput } from "../components/code-input";
+import { TipsSection } from "../components/tips-section";
+
+import { SceneKind } from "../types";
+import { useEmailRegisterStore, useScenesStore } from "../stores";
+
+export function InputEmailToRegister1() {
+	const goTo = useScenesStore(({ goTo }) => goTo);
+	const store = useEmailRegisterStore();
+	const refreshSize = useRefreshDynamicContainer();
+
+	// eslint-disable-next-line react-hooks/exhaustive-deps
+	React.useEffect(refreshSize, [store.isCodeSent]);
+
+	const verifyCode = React.useCallback(() => {
+		if (store.computed.canVerifyCode) {
+			store.verifyCode().then((isCodeValid) => {
+				if (isCodeValid) {
+					goTo(SceneKind.inputEmailToRegister2);
+				}
+			});
+		}
+	}, [store, goTo]);
+
+	return (
+		<>
+			<Header title="Register Email Account 1/2" />
+			<div data-animation="scale-fade-in" className="sm:w-362px p-24px">
+				<Field
+					title="Email Address"
+					icon={
+						<EmailIcon
+							className={classNames(
+								"transition",
+								store.emailErrorMsg ? "text-[#E65040]" : "text-[#FFB74D]"
+							)}
+						/>
+					}
+					className="mb-24px"
+					tips={<span className="text-[#E65040]">{store.emailErrorMsg}</span>}
+				>
+					<TextInput
+						name="email"
+						type="email"
+						spellCheck="false"
+						autoCorrect="off"
+						autoCapitalize="off"
+						autoComplete="email"
+						onBlur={store.validateEmail}
+						value={store.email}
+						onKeyDown={({ key }) => {
+							if (key === "Enter") {
+								store.sendCode();
+							}
+						}}
+						onChange={(e) => store.updateEmail(e.currentTarget.value)}
+						disabled={store.computed.isPending}
+						className="pr-108px"
+						rightSection={
+							<button
+								disabled={!store.computed.canSendCode}
+								className={classNames(
+									store.computed.isPending
+										? "cursor-progress"
+										: store.computed.canSendCode
+										? "cursor-pointer"
+										: "cursor-not-allowed",
+									!store.computed.canSendCode && "bg-opacity-50",
+									"transition whitespace-nowrap h-full w-100px flex items-center justify-center font-roboto font-500 text-14px text-white bg-[#6AD991] border-none rounded-r-12px"
+								)}
+								onClick={() => store.sendCode()}
+							>
+								{store.isCodeSent ? "Try Again" : "Verify"}
+								{store.codeResendCountdown > 0
+									? ` (${store.codeResendCountdown})`
+									: ""}
+							</button>
+						}
+					/>
+				</Field>
+
+				{store.isCodeSent && (
+					<TipsSection className="-mt-14px mb-24px">
+						{
+							"We've sent an email to verify your account, make sure it didn't wind up in your Junk Mail and enter the Verification Code"
```

Comment:
We've sent a verification email to you. Please check your email inbox and enter the Verification Code. If you didn't receive the email after a while, please check your spam or junk mail folder or contact us.